New package: msmpi - Microsoft MPI SDK

okhlybov opened the pull request:
Introduce the [Microsoft MPI](https://github.com/microsoft/Microsoft-MPI) SDK.
The package contains headers, import library and mpicc & friends compiler drivers necessary to build MPI-aware C/C++/FORTRAN code.

Alexpux commented on mingw-w64-msmpi/PKGBUILD:
```diff
@@ -0,0 +1,57 @@
+depends=("${MINGW_PACKAGE_PREFIX}-gcc" "${MINGW_PACKAGE_PREFIX}-gcc-fortran")
+options=('strip')
+
+source=('mpi.c')
```
Add all files that used for buid here

Alexpux commented on mingw-w64-msmpi/PKGBUILD:
```diff
@@ -0,0 +1,57 @@
+  # Import library
+  ${MINGW_PREFIX}/bin/dlltool -k -d "$rootdir/$CARCH/lib/msmpi.def" -l lib/libmsmpi.a
+  # Headers
+  cp $rootdir/include/* $rootdir/$CARCH/include/* include
```
Dont use rootdir, use srcdir instead everywhere

Alexpux commented on mingw-w64-msmpi/PKGBUILD:
```diff
@@ -0,0 +1,57 @@
+pkgdesc="Microsoft MPI SDK (mingw-w64)"
+arch=('any')
+url="https://docs.microsoft.com/en-us/message-passing-interface/microsoft-mpi"
+
```
Remove empty lines in header

Alexpux commented on mingw-w64-msmpi/PKGBUILD:
```diff
@@ -0,0 +1,57 @@
+options=('strip')
+
+source=('mpi.c')
+sha256sums=('SKIP')
```
when add all files to source run updpkgsums to generate chechsums

Alexpux commented on mingw-w64-msmpi/PKGBUILD:
```diff
@@ -0,0 +1,57 @@
+source=('mpi.c')
+sha256sums=('SKIP')
+
+rootdir=`pwd`
```
Remove this

okhlybov commented:
Fixed.

lharcke commented:
There's a number of different versions of MS-MPI available from the MS developer site, would it be possible to somehow include versioning info in this package, i.e. is this package targeting MS-MPI v7.x, v8.x, v9.x, or v10.x?  Things tend not to be compatible across major version numbers.

okhlybov commented:
The msmpi package is versioned. Currently it is built from the MS-MPI SDK 10.1.1 and this is reflected in the package's full name. Just to note this is a SDK and thus the MS-MPI runtime is required to be installed separately in order to acutally run the compiled code. The runtime's version does not need to match the package's version exactly. As MPI's API/ABI are pretty stable there is even a chance for the forward compatibility e.g. the code compiled with new msmpi will happily run with the older runtime provided no new features are used. According to the docs the were no API changes since v9.0 thus it can be assumed that the code built with current msmpi package is compatible with the MS-MPI runtime version 9 and upwards.

lharcke commented:
Yep, I should have done a `pacman -Ss msmpi` before commenting. Thanks.
